pkg/storage/unified/sql/backend.go

```diff
@@ -377,6 +396,36 @@ func (b *backend) create(ctx context.Context, event resource.WriteEvent) (int64,
 	return rv, nil
 }
 
+// isRowAlreadyExistsError checks if the error is the result of the row inserted already existing.
+//
+// On SQLite and Postgres, this is known as a UNIQUE constraint violation. On MySQL, it's known as a duplicate entry.
+// On Spanner, it's a gRPC ALREADY_EXISTS error.
+func isRowAlreadyExistsError(err error) bool {
```

Comment:
Should we have a general "error remapping handler"? 

I'd guess the need of remapping implementation specific to generic errors won't be restricted to this use case and we'd want to avoid implementation-specific errors to propagate upwards.

Reply:
I think it'd be helpful but we should probably not do _this_ for all rows. Even in the same function, I would say it's expected to have a duplicate on the resource but _not_ on the resource version (where it should instead cause a more fatal error).

Reply:
Can we move this check into the dialect? (available in `b.dialect`)

Reply:
I think that would make _more_ sense, if anything, yeah! I'll do that shortly